sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -564,6 +565,111 @@ public void testParDoWithSideInputs() {
   }
 
   @Test
+  @Category({ValidatesRunner.class, UsesImplicitSideInputs.class})
+  public void testParDoWithImplicitSideInputs() {
+    final PCollectionView<String> foo = pipeline
+        .apply("Create foo", Create.of("foo"))
+        .apply("Singleton", View.<String>asSingleton());
+
+    PCollection<String> output =
+        pipeline
+            .apply(Create.of(1, 2, 3))
+            .apply(
+                ParDo.of(
+                        new DoFn<Integer, String>() {
+                          @ProcessElement
+                          public void process(ProcessContext c) {
+                            c.output(foo.get() + " " + c.element());
+                          }
+                        })
+                    .withSideInputs(foo));
+
+    PAssert.that(output).containsInAnyOrder("foo 1", "foo 2", "foo 3");
+
+    pipeline.run();
+  }
+
+  @Test
+  @Category({ValidatesRunner.class, UsesImplicitSideInputs.class})
+  public void testParDoWithImplicitSideInputsUndeclared() {
+    final PCollectionView<String> foo = pipeline
+        .apply("Create foo", Create.of("foo"))
+        .apply("As singleton", View.<String>asSingleton());
+
+    pipeline
+        .apply(Create.of(1, 2, 3))
+        .apply(
+            ParDo.of(
+                new DoFn<Integer, String>() {
+                  @ProcessElement
+                  public void process(ProcessContext c) {
+                    c.output(foo.get() + " " + c.element());
+                  }
+                }));
+
+    thrown.expectMessage("undeclared side input view");
+    thrown.expectMessage("Create foo");
+    thrown.expectMessage("As singleton");
+    thrown.expectMessage(".withSideInputs()");
+
+    pipeline.run();
+  }
+
+  @Test
+  @Category({ValidatesRunner.class, UsesImplicitSideInputs.class})
```

Comment:
To make this a ValidatesRunner test for a failure scenario catch and convert the exception into a stack trace and match against the contents of the string using PAssert. There is no expectation that a test pipeline runner can accurately rethrow the failure reason to the process that submitted the job.

Ditto on the finishBundle test as well.